File: codes/OOP/cpp/inheritance.cpp
//Defining a class in terms of another class. This helps in code reuse.

#include<bits/stdc++.h>
using namespace std;
class Shape
{
   protected:
      int width, height;
   public:
      void setWidth(int w)
      {
         width = w;
      }
      void setHeight(int h)
      {
         height = h;
      }
};
class Rectangle: public Shape
{
   public:
      int getArea()
      {
         return (width * height);
      }
};

int main()
{
   Rectangle Rect;
   Rect.setWidth(5);
   Rect.setHeight(7);
   cout << Rect.getArea() << endl;
   return 0;
}

File: codes/OOP/java/Inheritance.java
/**
 * Created by gaurav on 6/2/16.
 */
class Shape {
    protected int height, width;
    public void setHeight(int h) {
        height = h;
    }
    public void setWidth(int w) {
        width = w;
    }
}
class Rectangle extends Shape {
    public void printArea() {
        System.out.println(width*height);
    }
}
public class Inheritance {
    public static void main(String args[]) {
        Rectangle rectangle = new Rectangle();
        rectangle.setHeight(5);
        rectangle.setWidth(4);
        rectangle.printArea();
    }

}